### PR TITLE
[ADD] website_variants_extra: phantom test product comment qty

### PR DESCRIPTION
**Pull request by gmlp**
Dummy PR: Vauxoo/yoytec#803

### CODE REVIEW

File: website_variants_extra/static/src/js/test_comments_qty.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        name: 'Test Product Name with Variants',
+        path: '/shop/product/ipad-retina-display-5',
+        mode: 'test',
+        steps: [
+            {
+                title:     "Post first comment",
+                onload:    function(Tour){
+                               $('#comment textarea').text('1st Comment');
+                               $('#comment .fa-star-o:last').trigger('click');
+                               $('#comment .btn').trigger('click');
```
Comment (oscarolar) [inline website_variants_extra/static/src/js/test_comments_qty.js]: Only when a form has a `preventDefault` or the action of the button is given by a `.click` event on JS is necessary to trigger the click event, when is a button on a form you must use `element` key to trigger its `post` method, the last trigger here is wrong. please replace with the right `element` key as follows: `element: "form a:contains(Post)",`

File: website_variants_extra/static/src/js/test_comments_qty.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+                            },
+                element:   '#comment textarea',
+            },
+            {
+                title:     "Post second comment",
+                onload:    function(Tour){
+                               $('#comment textarea').text('2nd Comment');
+                               $('#comment .fa-star-o:last').trigger('click');
+                               $('#comment .btn').trigger('click');
```
Comment (oscarolar) [inline website_variants_extra/static/src/js/test_comments_qty.js]: same here as above

File: website_variants_extra/tests/test_comments_qty.py [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+@openerp.tests.common.post_install(True)
+class TestUi(openerp.tests.HttpCase):
+
+    def test_01_sale_product_name_admin(self):
+        self.phantom_js("/",
+                        "openerp.Tour.run('shop_test_comment_qty',\
+                        'test')",
+                        "openerp.Tour.tours.shop_test_comment_qty",
+                        login='admin')
+
```
Comment (oscarolar) [inline website_variants_extra/tests/test_comments_qty.py]: test with demo and portal  too `login=demo` and `login=portal` too.

File: website_variants_extra/tests/test_comments_qty.py [hunk 4]
```diff
@@ -0,0 +1,15 @@
+# coding: utf-8
+import openerp.tests
+
+
+@openerp.tests.common.at_install(False)
+@openerp.tests.common.post_install(True)
+class TestUi(openerp.tests.HttpCase):
+
+    def test_01_sale_product_name_admin(self):
```
Comment (oscarolar) [inline website_variants_extra/tests/test_comments_qty.py]: use the correct name here `test_01_product_comments_admin`

**Comment by oscarolar** [comment]
where is the dummy build of this PR?

**Comment by gmlp** [comment]
Dummy PR: https://github.com/Vauxoo/yoytec/pull/803

**Comment by oscarolar** [comment]
:+1: @nhomar any thougts?

File: website_variants_extra/static/src/js/test_comments_qty.js [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+                onload:    function(Tour){
+                               $('#comment textarea').text('2nd Comment');
+                               $('#comment .fa-star-o:last').trigger('click');
+                            },
+                waitFor:   '#comments-list:first:contains(1st)',
+                element:  'form a:contains(Post)',
+            },
+            {
+                title: 'check comment qty',
+                waitFor: '#no-reviews:contains(1)',
```
Comment (nhomar) [inline website_variants_extra/static/src/js/test_comments_qty.js]: Esto debería probar la "cantidad" no solo que exista el elemento.

Propongo:

Vía xml (no phantom) crea N comentarios dummy (digamos 4 para que no sea solo 2) entonces aquí simplemente sería:

Paso 1:- Verifico que encuentro el tag y que dice "4".
Paso 2:- ta cual lo que tienes aquí creo dos comentarios.
Paso 3:- Tal cual tienes aquí verifico la etiqueta está pero ahora dice "6".

Saludos.

Comment (oscarolar) [inline website_variants_extra/static/src/js/test_comments_qty.js]: @nhomar 
Esto debería probar la "cantidad" no solo que exista el elemento.:
it acctually tests "How many published comments are on the product"
`waitFor: '#no-reviews:contains(1)',` <-- this code reffers to a label 'review {#no-reviews} comments' not if it exists, maybe you didn't quite understood the test, the only difference between your approach and what is done is the number, 1 or 6.

Best regards.

Comment (gmlp) [inline website_variants_extra/static/src/js/test_comments_qty.js]: @nhomar,  the quantity it's verified, 
the test behavior it's the following:
1. Create a Comment
2. Publish the first comment
3. Create another comment
4. Verify that only published comment was taken in acconunt.
Best regards.

Comment (gmlp) [inline website_variants_extra/static/src/js/test_comments_qty.js]: @nhomar , asked changes done.

**Pull request merged**